core/src/impl/Kokkos_ViewLayoutTiled.hpp

```diff
@@ -659,6 +659,91 @@ struct ViewOffset<
                                : 0) {}
 };
 
+// FIXME Remove the out-of-class definitions when we require C++17
+#define KOKKOS_ITERATE_VIEW_OFFSET_ENABLE                                      \
+  typename std::enable_if<((Dimension::rank <= 8) && (Dimension::rank >= 2) && \
```

Comment:
Minor:  Now that C++14 is the base, can we use `enable_if_t`?

Minor:  I would expect to see the two `Dimension::rank` checks reversed.